perfact/zodbsync/zodbsync.py

```diff
@@ -55,74 +56,69 @@ def mod_format(data=None, indent=0, as_list=False):
     '''
 
     def str_repr(val):
-        '''Generic string representation of a value.'''
-        return str((val,))[1:-2]
-
-    def split_longlines(lines, maxlen=100, threshold=140):
-        '''Split a list of strings into a longer list of strings, but each
-        with lines no longer than <threshold>, split at <maxlen>.'''
-        index = 0
-        while True:
-            if len(lines[index]) > threshold:
-                remainder = lines[index][maxlen:]
-                lines[index] = lines[index][:maxlen]
-                lines.insert(index+1, remainder)
-            index += 1
-            if index == len(lines):
-                break
-        return lines
-
-    output = []
+        '''Generic string representation of a value'''
+
+        if isinstance(val, list):
+            return '[%s]' % ', '.join(str_repr(item) for item in val)
+        elif isinstance(val, tuple):
+            fmt = '(%s,)' if len(val) == 1 else '(%s)'
+            return fmt % ', '.join(str_repr(item) for item in val)
+
+        if PY2 and isinstance(val, (bytes, unicode)):
+            '''
+            One might assume that a most stringent representation would always
+            prefix the value with either b or u to denote bytes or unicode.
+            However, properties that were stored as bytes in Python2 (like
+            title) usually have become unicode in Python3. In a default PerFact
+            installation, these properties were always *meant* to be UTF-8
+            encoded text. So the best representation is to store them without
+            prefix and with as few escapes as possible (so no \xc3\xbc, but
+            simply ü).  The only characters that need to be escaped are \n, \r,
+            \\ and \' (because we enclose the expression in '').
+            To keep the diff to older versions smaller, we also check if there
+            is a ' but no " inside, switching the enclosing quotation marks.
+            '''
```

Comment:
Why should we reduce the number of escaped characters (and use the eightth bit)? Compare this (called from an UTF-8 terminal):

    Python2
    >>> a = 'abcdöäü'
    >>> print(repr(a))
    'abcdöäü'

    Python3 (and Python2 from a script)
    >>> a = 'abcdöäü'
    >>> print(repr(a))
    'abcd\xc3\xb6\xc3\xa4\xc3\xbc'

It looks like `repr()` in Python3 (and Python2 from scripts) naturally uses 7 bit ascii to represent strings (as I would expect it to), only Python2 seems to use UTF-8 if the terminal is UTF-8. But from a script, both Python2 and Python3 consistently use 7 bit ascii to represent  all strings.

This was the original intention of the string representation: To use only 7 bits. In the face of this, I would like to discuss the motivation behind this change.

Reply:
I confused Python2 and Python3 in the example above. Sorry. Python3 does something completely different with `repr()`, so that explains a lot.